src/services/user.js

```
@@ -177,7 +183,276 @@ export async function deleteUser(userId) {
   }
 }
 
+/**
+ * Check if user exists and return user data, or null if not found
+ * @param {string} userId - User ID to check
+ * @returns {Promise<Partial<UserEntitlementDocument>|null>} User data or null if not found
+ */
+async function findExistingUser(userId) {
+  try {
+    return await get(userId)
+  } catch {
+    return null
```

Bit dangerous. Better to let it exceptions bubble up so it gets interpreted as an internal server error, rather than masking errors (e.g. if the DB was down). You'd need to explicitly check if the resource is not found in this error and let the other exceptions bubble up

---

Yeah agreed